src/Geometry/Transform.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures #-}
+module Geometry.Transform
+( Transform(..)
+, mkTranslation
+, mkScale
+, mkRotation
+, apply
+, tmap
+, (>>>)
+) where
+
+import Control.Category
+import Control.Lens ((&), (.~))
+import Foreign.Storable
+import GL.Type as GL
+import GL.Uniform
+import Linear.Exts
+import Prelude hiding ((.))
+
+newtype Transform c (a :: * -> *) (b :: * -> *) = Transform { getTransform :: M44 c }
+  deriving (Show, Storable, GL.Type, Uniform)
+
+instance Num c => Category (Transform c) where
+  id = Transform identity
+  Transform a . Transform b = Transform (b !*! a)
```

Comment:
This is pretty important to scaling things out nicely. This gives us names for each of the different coordinate systems we operate in, and enforces that we’re using the right transitions for the right spaces in the right order to end up at the right place.

It’s also quite efficient because representationally it’s just a `newtype` around a matrix.